src/heretic/model.py

```diff
@@ -105,10 +114,19 @@ def reload_model(self):
         self.model = None
         empty_cache()
 
+        max_memory = (
+            {
+                int(k) if k.isdigit() else k: v
+                for k, v in settings.max_memory.items()
+            }
+            if settings.max_memory
+            else None
+        )
```

Comment:
No need to duplicate this code. You should be able to reuse the `max_memory` value stored in the model object. See the equivalent code for `dtype` a few lines up.